README.markdown

```diff
@@ -650,233 +976,293 @@ Looks up a variable in a remote namespace.
 
 For example:
 
-  ~~~
-  $foo = getvar('site::data::foo')
-  # Equivalent to $foo = $site::data::foo
-  ~~~
+```puppet
+$foo = getvar('site::data::foo')
+# Equivalent to $foo = $site::data::foo
+```
 
 This is useful if the namespace itself is stored in a string:
 
-  ~~~
-  $datalocation = 'site::data'
-  $bar = getvar("${datalocation}::bar")
-  # Equivalent to $bar = $site::data::bar
-  ~~~
+```puppet
+$datalocation = 'site::data'
+$bar = getvar("${datalocation}::bar")
+# Equivalent to $bar = $site::data::bar
+```
 
 *Type*: rvalue.
 
 #### `glob`
 
-Dir#glob wrapper that accepts a string or an array of strings of path patterns.
-Returns an array of strings of matched paths.
+Returns an array of strings of paths matching path patterns.
 
-  ~~~
-  $confs = glob(['/etc/**/*.conf', '/opt/**/*.conf'])
-  ~~~
+Arguments: A string or an array of strings specifying path patterns.
+
+```puppet
+$confs = glob(['/etc/**/*.conf', '/opt/**/*.conf'])
+```
 
 *Type*: rvalue.
 
 #### `grep`
 
-Searches through an array and returns any elements that match the provided regular expression. For example, `grep(['aaa','bbb','ccc','aaaddd'], 'aaa')` returns ['aaa','aaaddd']. *Type*: rvalue.
+Searches through an array and returns any elements that match the provided regular expression.
+
+For example, `grep(['aaa','bbb','ccc','aaaddd'], 'aaa')` returns ['aaa','aaaddd'].
+
+*Type*: rvalue.
 
 #### `has_interface_with`
 
-Returns a boolean based on kind and value:
+Returns a Boolean based on kind and value:
+
   * macaddress
   * netmask
   * ipaddress
   * network
 
 *Examples:*
 
-  ~~~
-  has_interface_with("macaddress", "x:x:x:x:x:x")
-  has_interface_with("ipaddress", "127.0.0.1")    => true
-  ~~~
+```puppet
+has_interface_with("macaddress", "x:x:x:x:x:x")
+has_interface_with("ipaddress", "127.0.0.1")    => true
+```
 
 If no kind is given, then the presence of the interface is checked:
 
-  ~~~
-  has_interface_with("lo")                        => true
-  ~~~
+```puppet
+has_interface_with("lo")                        => true
+```
 
 *Type*: rvalue.
 
 #### `has_ip_address`
 
-Returns 'true' if the client has the requested IP address on some interface. This function iterates through the `interfaces` fact and checks the `ipaddress_IFACE` facts, performing a simple string comparison. *Type*: rvalue.
+Returns `true` if the client has the requested IP address on some interface.
```

Comment:
I would keep the second sentence here as it's important to know where it's checking against and how. Otherwise you'd have to look at the function code itself to find out.